components/script/dom/vrdisplay.rs

```diff
@@ -0,0 +1,618 @@
+    }
+
+    // https://w3c.github.io/webvr/#dom-vrdisplay-ispresenting
+    fn IsPresenting(&self) -> bool {
+        self.presenting.get()
+    }
+
+    // https://w3c.github.io/webvr/#dom-vrdisplay-capabilities
+    fn Capabilities(&self) -> Root<VRDisplayCapabilities> {
+        Root::from_ref(&*self.capabilities.get())
+    }
+
+    // https://w3c.github.io/webvr/#dom-vrdisplay-stageparameters
+    fn GetStageParameters(&self) -> Option<Root<VRStageParameters>> {
+        self.stage_params.get().map(|s| Root::from_ref(&*s))
+    }
+
+    // https://w3c.github.io/webvr/#dom-vrdisplay-geteyeparameters
+    fn GetEyeParameters(&self, eye: VREye) -> Root<VREyeParameters> {
+        match eye {
+            VREye::Left => Root::from_ref(&*self.left_eye_params.get()),
+            VREye::Right => Root::from_ref(&*self.right_eye_params.get())
+        }
+    }
+
+    // https://w3c.github.io/webvr/#dom-vrdisplay-displayid
+    fn DisplayId(&self) -> u32 {
+        self.display.borrow().0.display_id as u32
+    }
+
+    // https://w3c.github.io/webvr/#dom-vrdisplay-displayname
+    fn DisplayName(&self) -> DOMString {
+        DOMString::from(self.display.borrow().0.display_name.clone())
+    }
+
+    // https://w3c.github.io/webvr/#dom-vrdisplay-getframedata-framedata-framedata
+    fn GetFrameData(&self, frameData: &VRFrameData) -> bool {
+        // If presenting we use a synced data with compositor for the whole frame
+        if self.presenting.get() {
+            if self.frame_data_status.get() == VRFrameDataStatus::Waiting {
+                self.sync_frame_data();
+            }
+            frameData.update(& self.frame_data.borrow().0);
+            return true;
+        }
+
+        // If not presenting we fetch inmediante VRFrameData
+        if let Some(wevbr_sender) = self.webvr_thread() {
+            let (sender, receiver) = ipc::channel().unwrap();
+            wevbr_sender.send(WebVRMsg::GetFrameData(self.global().pipeline_id(),
+                                                     self.get_display_id(),
+                                                     self.depth_near.get(),
+                                                     self.depth_far.get(),
+                                                     sender)).unwrap();
+            return match receiver.recv().unwrap() {
+                Ok(data) => {
+                    frameData.update(&data);
+                    true
+                },
+                Err(e) => {
+                    error!("WebVR::GetFrameData: {:?}", e);
+                    false
+                }
+            };
+        }
+
+        false
+    }
+
+    // https://w3c.github.io/webvr/#dom-vrdisplay-getpose
+    fn GetPose(&self) -> Root<VRPose> {
+        VRPose::new(&self.global(), &self.frame_data.borrow().0.pose)
+    }
+
+    // https://w3c.github.io/webvr/#dom-vrdisplay-resetpose
+    fn ResetPose(&self) -> () {
+        if let Some(wevbr_sender) = self.webvr_thread() {
```

Comment:
It shouldn't happen unless we want to allow something like stopping/disabling WebVR when Servo is already running (without requiring to restart it).

What's the guideline for this kind of cases on Servo, a panic or a console error?